browser/ai_chat/ai_chat_ui_browsertest.cc

```diff
@@ -0,0 +1,224 @@
+#if BUILDFLAG(ENABLE_TEXT_RECOGNITION)
+    command_line->AppendSwitch(::switches::kEnablePixelOutputInTests);
+#endif
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+    InProcessBrowserTest::TearDownInProcessBrowserTestFixture();
+  }
+
+  content::WebContents* GetActiveWebContents() {
+    return browser()->tab_strip_model()->GetActiveWebContents();
+  }
+
+  void NavigateURL(const GURL& url) {
+    ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+    ASSERT_TRUE(WaitForLoadStop(GetActiveWebContents()));
+  }
+
+  void CreatePrintPreview(ai_chat::AIChatUIPageHandler* handler) {
+#if BUILDFLAG(ENABLE_PRINT_PREVIEW)
+    handler->MaybeCreatePrintPreview();
+#endif
+  }
+
+  ai_chat::AIChatUIPageHandler* OpenAIChatSidePanel() {
+    auto* side_panel_ui = SidePanelUI::GetSidePanelUIForBrowser(browser());
+    side_panel_ui->Show(SidePanelEntryId::kChatUI);
+    auto* browser_view = BrowserView::GetBrowserViewForBrowser(browser());
+    auto* side_panel = browser_view->unified_side_panel();
+    auto* ai_chat_side_panel =
+        side_panel->GetViewByID(SidePanelWebUIView::kSidePanelWebViewId);
+    if (!ai_chat_side_panel) {
+      return nullptr;
+    }
+    auto* side_panel_web_contents =
+        (static_cast<views::WebView*>(ai_chat_side_panel))->web_contents();
+    if (!side_panel_web_contents) {
+      return nullptr;
+    }
+    content::WaitForLoadStop(side_panel_web_contents);
+
+    auto* web_ui = side_panel_web_contents->GetWebUI();
+    if (!web_ui) {
+      return nullptr;
+    }
+    auto* ai_chat_ui = web_ui->GetController()->GetAs<AIChatUI>();
+    if (!ai_chat_ui) {
+      return nullptr;
+    }
+    return ai_chat_ui->GetPageHandlerForTesting();
+  }
+
+  void FetchPageContent(const base::Location& location,
+                        ai_chat::AIChatTabHelper* helper,
+                        std::string_view expected_text) {
+    SCOPED_TRACE(testing::Message() << location.ToString());
+    base::RunLoop run_loop;
+    helper->GetPageContent(
+        base::BindLambdaForTesting(
+            [&run_loop, expected_text](std::string text, bool is_video,
+                                       std::string invalidation_token) {
+              EXPECT_FALSE(is_video);
+              EXPECT_EQ(text, expected_text);
+              run_loop.Quit();
+            }),
+        "");
+    run_loop.Run();
+  }
+
+ protected:
+  net::test_server::EmbeddedTestServer https_server_;
+
+ private:
+  content::ContentMockCertVerifier mock_cert_verifier_;
+};
+
+IN_PROC_BROWSER_TEST_F(AIChatUIBrowserTest, PrintPreview) {
+  browser()->window()->SetContentsSize(gfx::Size(800, 600));
```

Comment:
Could add a comment here that the test creates divs of this size. Also I think this can be moved to `SetUpOnMainThread` and you can remove the 3 calls at the start of the test files.

Reply:
addressed and squashed.